### ISSUE TITLE
Loss decreasing slowly while moving the GO flag out of ground truth.

### ISSUE THREAD

**Issue by bidai541** [emedvedev/attention-ocr#105]
I found that the GO flags have been remove while caculating the loss function, but why? In my experiments, the speed of convergency including the GO flags is faster.
May be a naive question, thanks.

-------------------------------------------------------------------------------------------------
UPDATE
It's wrong to include GO flag while caculating the seq2seq loss while decoder is decode the next char appending current one. 
I changed some details. First, the initial decoder_input while inference should be go_index. Second, I change the decode function with attention.

```
    for i, inp in enumerate(decoder_inputs):
      if i > 0:
        tf.get_variable_scope().reuse_variables()
      if loop_function is not None and prev is not None:
        with tf.variable_scope("loop_function", reuse=True):
          inp = loop_function(prev, i)
      input_size = attn_num_hidden
      if i == 0 and initial_state_attention:
        with tf.variable_scope(tf.get_variable_scope(),
                               reuse=True):
          attns, attn_weights = attention(state)
      else:
        attns, attn_weights = attention(state)
        attention_weights_history.append(attn_weights)
      x = linear([inp] + attns, input_size, True)
      cell_output, state = cell(x, state)
      with tf.variable_scope("AttnOutputProjection"):
        output = linear(cell_output, output_size, True)
      if loop_function is not None:
        prev = output
      outputs.append(output)
```
Decoding the first char using attention context vector by initial state combine the embedding vector instead of using initial_state only. 
Fiannaly I trained the model by joining CTC loss and seq2seq loss together, result sames better.

For reference only, thanks again!

**Comment by emedvedev** [comment]
I honestly don't remember if there was a particular reason behind it, but please do try to experiment, and if you see better speed with the same result, submit a PR. :)
